Core/Core/Features/Files/View/FileList/FileListViewController.swift

```diff
@@ -116,6 +127,14 @@ public class FileListViewController: ScreenViewTrackableViewController, ColoredN
             self.tableView.contentOffset.y = self.searchBar.frame.height
         }
 
+        studentAccessInteractor?
+            .isRestricted()
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] isRestricted in
+                self?.isRestricted = isRestricted
+            }
```

Comment:
nit: Instead of sink and weak self:
```suggestion
            .assign(to: \.isRestricted, on: self, ownership: .weak)
```